### PR TITLE
Fix `Properties.store(OutputStream os, String comment)`

### LINKED ISSUE #23: `Properties#store(OutputStream)` doesn't flush the output to the `OutputStream`
The following test fails: 

```java
    @Test
    void testStoreOutputStream() throws IOException, URISyntaxException {
        Path f = getResource("/test.properties");
        Properties p = Properties.loadProperties(f);
        ByteArrayOutputStream os = new ByteArrayOutputStream();
        p.store(os);
        assertThat(os.toString()).isEqualTo(readAll(f));
    }
```

Probably the OutputStreamWriter isn't flushing to the underlying OutputStream?

### PR DESCRIPTION
**Pull request by gastaldi**
This fixes the store method taking an `OutputStream` by flushing and closing the writer created to wrap it.

~~It also introduces ApprovalTests (https://approvaltests.com/) to open a diff window when comparing Strings and File contents~~

- Fixes #23

### CODE REVIEW

**Comment by quintesse** [comment]
Ok, first, good catch, the docs of `Properties.load()` does indeed say the stream should be flushed and the underlying stream _not_ closed:

> After the entries have been written, the output stream is flushed. The output stream remains open after this method returns.

(closing the writer doesn't close the stream, right? I never know and the docs are not clear)

Second... introducing a new way of testing? Awesome... are you going to migrate the rest of the tests as well? 😁 😛

**Comment by gastaldi** [comment]
> (closing the writer doesn't close the stream, right? I never know and the docs are not clear)

Actually the writer flushes and then closes the stream. Maybe we should only flush it instead of closing in a try-with-resources block, I'll check.

> Second... introducing a new way of testing? Awesome... are you going to migrate the rest of the tests as well?

I was thinking of leaving the other tests that as an exercise for the reader 😉

File: src/main/java/org/codejive/properties/Properties.java [hunk 1]
```diff
@@ -938,6 +940,7 @@ public void store(Writer writer, String... comment) throws IOException {
             writer.write(pos.raw());
             pos.next();
         }
+        writer.flush();
```
Comment (gastaldi) [inline src/main/java/org/codejive/properties/Properties.java:943]: This was the missing piece

**Comment by quintesse** [comment]
> I was thinking of leaving the other tests as an exercise for the reader

Hahaha, ok, can I then ask a favor? Could you split the PR in two, one for this fix with a test that follows the same pattern as the other tests and then a new PR showing a single test converted to using that approvals library? I'll then use that PR as a basis for migrating all file-based tests to that system in the future.

**Review by quintesse — CHANGES_REQUESTED** [review]
Split into two PRs

**Comment by gastaldi** [comment]
Created https://github.com/codejive/java-properties/pull/26 (I'll rebase once this PR is merged)

**Review by quintesse — APPROVED** [review]

**Comment by quintesse** [comment]
Thanks man! :-)

**Pull request merged**